File: src/main/java/eu/valawai/mov/events/TopologyAction.java
/*
  Copyright 2023 UDT-IA, IIIA-CSIC

  Use of this source code is governed by GNU General Public License version 3
  license that can be found in the LICENSE file or at
  https://opensource.org/license/gpl-3-0/
*/

package eu.valawai.mov.events;

/**
 * The type of action to do over the topology.
 *
 * @author VALAWAI
 */
public enum TopologyAction {

	/**
	 * Enable the connection between components.
	 */
	ENABLE,

	/**
	 * Disable the connection between components.
	 */
	DISABLE;

}

File: src/main/webui/src/app/main/topology/design/editor/topology.action.ts
/*
  Copyright 2022-2026 VALAWAI

  Use of this source code is governed by GNU General Public License version 3
  license that can be found in the LICENSE file or at
  https://opensource.org/license/gpl-3-0/
*/

import { TopologyEditorService } from "./topology.service";


/**
 * An action that chnage the topology.
 */
export interface TopologyEditorAction {

	/**
	 * Undo the topology change action.
	 */
	undo(service: TopologyEditorService): void;

	/**
	 * Redo the topology change action.
	 */
	 redo(service: TopologyEditorService): void;
}
